src/Microsoft.AspNet.Mvc.Core/ViewComponents/DefaultViewComponentInvoker.cs

```diff
@@ -34,6 +38,12 @@ public void Invoke([NotNull] ViewComponentContext context)
                 context.Arguments);
             if (method == null)
             {
+                _logger.LogError(
+                    "Could not find a method with name '{ViewComponentMethodName}' on view " +
+                    "component '{ViewComponentName}'.",
+                    ViewComponentMethodSelector.SyncMethodName,
+                    context.ViewComponentDescriptor.FullName);
```

Comment:
At this point the guidance is not to log where we throw an exception that will be go unhanded by the framework.